singer_sdk/mapper.py

```diff
@@ -666,6 +666,9 @@ def register_raw_stream_schema(
             ]
 
         for stream_map_key, stream_def in self.stream_maps_dict.items():
+            stream_def = (
+                dict(stream_def) if isinstance(stream_def, dict) else stream_def
+            )
```

Comment:
Would this work too?

```suggestion
            stream_def = (
                stream_def.copy() if isinstance(stream_def, dict) else stream_def
            )
```